website/docs/core-concepts/connecting-to-data-sources/connecting-to-databases.md

```diff
@@ -5,59 +5,76 @@ description: >-
+* Click **Test** to verify that Appsmith can connect to your database using the details you provided.
+* Rename and save your datasource.
+
+It's important to note that the configuration of the datasource aren't be updated/saved unless the "save" button is clicked. Once your datasource has been added successfully, a success pop-up appears at the top. 
+
+### Mock database
+
+
+Appsmith provides two sample datasets to help you connect the data and test the platform. 
 
-**18.223.74.85** and **3.131.104.27** are the IP addresses of the Appsmith cloud instances that need to be whitelisted.
+1. A sample MongoDB database with a collection called "movies."
+2. A sample PostgreSQL database with a table called "users."
+
+:::note
+The mock databases are public and shared by all users, which may result in mangled data. To refresh the data, the databases are automatically **reset every day**, so any updates made to these databases are temporary.
 :::
 
-Once the appsmith instance has been [whitelisted](../../learning-and-resources/how-to-guides/aws-whitelist.md):
+<VideoEmbed host="youtube" videoId="TrV8h_Dvhbg" title="Using A Sample Database " caption="How to use mock database"/>
 
-1. Click on **+** next to **Datasources**.
-2. You’ll see a list of Datasources that Appsmith can connect to.
-3. Choose a Database from the [supported list](connecting-to-databases.md#supported-databases)
-4. Provide the configuration details required to connect to your database. You may need to contact your database admin to [whitelist appsmith cloud](../../learning-and-resources/how-to-guides/aws-whitelist.md)
-5. Click on **Test** to verify that Appsmith is able to connect to your database using the details provided by you.
-6. **Save** your **Datasources**.
+* Click on **+** next to Datasources.
+* Select the mock **users** PostgreSQL database.
+* You can create queries by selecting the **New API +** button available on the **datasource** page.
+* Write the query in the query editor and click the **Run** button to execute it. For example:
 
-![](</img/create_ds.gif>)
+```sql
+SELECT * FROM users;
+```
+The select query retrieves the data from the users table. You can display the query results in a Table widget by [binding the data](/core-concepts/data-access-and-binding/displaying-data-read#displaying-data-in-a-widget) in the Table Data property.
 
-:::tip
+:::note
 Databases configured within one page of an app can be accessed by developers anywhere within the workspace.
 :::
 
-## General Notes
+## General notes
+Appsmith establishes a connection pool with the database server when you first connect the database to your app. This allows subsequent queries to be executed efficiently. If an idle connection is closed by the database server, Appsmith creates a new connection for the next query.
+
+For certain database plugins such as PostgreSQL and MySQL, Appsmith maintains a connection pool because multiple queries can't be run on a single connection.
```

Comment:
(1) Should we mention other plugins too that have connection pool - Redis, MsSQL, Redshift ? 
(2) Can you please replace `maintains a connection pool` with `explicitly creates and maintains a connection pool` ? - since for few plugins that underlying driver might do it implicitly. 
(3) Can you please replace `because multiple queries can't be run on a single connection` with `because usually a single connection is not designed to handle multiple simultaneous queries`

Reply:
done